app/controllers/staff/communal_defects_controller.rb

```diff
@@ -77,6 +82,10 @@ def actual_completion_date
     params.fetch(:actual_completion_date, {}).permit(:day, :month, :year)
   end
 
+  def created_at
+    params.fetch(:created_at, {}).permit(:day, :month, :year)
+  end
```

Comment:
What was the motivation for calling this `created_at` rather than `creation_date` in line with the other user entered dates? I think that would avoid clashes with the Rails default field's name.

Reply:
Also, `created_at` is more about when the Rails model was created, which might not match the user entered data.

Reply:
Great question, there is a lot of functionality that hangs off the `created_at` field, and this ticket is about letting users have a say in when this is, effectively overriding the default setting.

The alternative would have been to create a new field (like `creation_date`) then rewrite a lot of logic to use this new field. This would have been fine and it feels weird in the forms to have `created_at` but for everything else in the system it makes a lot more sense and is more Railsy (imo) that things stay as `created_at`.

Hope that makes sense.

Reply:
Yup. That does clear some things up. I'm on the fence about whether or not I agree, but you're closer to it, so do what you think's best. I'll take a proper look now, then.

Reply:
Happy to chat through IRL and more than happy to be proved wrong.